pages/dao/[symbol]/proposal/components/instructions/Raydium/RemoveLiquidityFromPool.tsx

```diff
@@ -0,0 +1,120 @@
+import { createRemoveLiquidityInstruction } from '@tools/sdk/raydium/createRemoveLiquidityInstruction'
+import { fetchLiquidityPoolData } from '@tools/sdk/raydium/helpers'
+import { liquidityPoolKeysList } from '@tools/sdk/raydium/poolKeys'
+import { notify } from '@utils/notifications'
+import { RemoveLiquidityRaydiumForm } from '@utils/uiTypes/proposalCreationTypes'
+
+import SelectOptionList from '../../SelectOptionList'
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+import { uiAmountToNativeBN } from '@tools/sdk/units'
+
+const RaydiumRemoveLiquidityFromPool = ({
+  index,
+  governedAccount,
+}: {
+  index: number
+  governedAccount?: GovernedMultiTypeAccount
+}) => {
+  const [lpMintInfo, setLpMintInfo] = useState<{
+    balance: number
+    decimals: number
+  } | null>(null)
+
+  const {
+    form,
+    connection,
+    formErrors,
+    handleSetForm,
+  } = useInstructionFormBuilder<RemoveLiquidityRaydiumForm>({
+    index,
+    initialFormValues: {
+      governedAccount,
+      liquidityPool: '',
+      amountIn: 0,
+    },
+    schema: yup.object().shape({
+      governedAccount: yup
+        .object()
+        .nullable()
+        .required('Program governed account is required'),
+      liquidityPool: yup.string().required('Liquidity Pool is required'),
+      amountIn: yup
+        .number()
+        .moreThan(0, 'Amount for LP token should be more than 0')
+        .required('Amount for LP token is required'),
+    }),
+    buildInstruction: async function () {
+      if (!form.liquidityPool || !lpMintInfo) {
+        throw new Error('missing field in instruction form')
+      }
+
+      return createRemoveLiquidityInstruction(
+        new PublicKey(governedAccount!.governance.pubkey),
+        jsonInfo2PoolKeys(liquidityPoolKeysList[form.liquidityPool]),
+        uiAmountToNativeBN(form.amountIn, lpMintInfo.decimals)
+      )
+    },
+  })
+
+  useEffect(() => {
+    async function fetchLpMintInfo() {
+      try {
+        const { maxBalance, decimals } = await fetchLiquidityPoolData({
+          governanceKey: governedAccount?.governance.pubkey,
+          lp: form.liquidityPool,
+          connection: connection.current,
+        })
+        setLpMintInfo({ balance: maxBalance, decimals })
+      } catch (e) {
+        notify({
+          type: 'error',
+          message: 'Could not fetch LP Account',
+          description: `${form.liquidityPool} LP Token Account could not be found for the selected Governance`,
+        })
+      }
+    }
+    fetchLpMintInfo()
+  }, [governedAccount?.governance.pubkey, form.liquidityPool])
+
+  return (
+    <>
+      <Select
+        label="Raydium Liquidity Pool"
+        value={form.liquidityPool}
+        placeholder="Please select..."
+        onChange={(value) =>
+          handleSetForm({ value, propertyName: 'liquidityPool' })
+        }
+        error={formErrors['liquidityPool']}
+      >
+        <SelectOptionList list={Object.keys(liquidityPoolKeysList)} />
```

Comment:
move to its own variable outside of the component, always try to have stable references when possible (here the list is re-computed at each render, which doesn't really matter here but still)